presto-spi/src/main/java/com/facebook/presto/spi/block/VariableWidthBlockBuilder.java

```diff
@@ -275,6 +276,12 @@ public void reset(BlockBuilderStatus blockBuilderStatus)
         updateArraysDataSize();
     }
 
+    @Override
+    public BlockBuilder newBlockBuilderLike(BlockBuilderStatus blockBuilderStatus)
+    {
+        return new VariableWidthBlockBuilder(blockBuilderStatus, calculateBlockResetSize(positions), EXPECTED_BYTES_PER_ENTRY);
```

Comment:
Instead of using the constant `EXPECTED_BYTES_PER_ENTRY`, use something line `(getOffset(positions) - getOffset(0)) / positions`